cache/key_test.go

```diff
@@ -28,10 +32,107 @@ func TestWithVersionSuffix(t *testing.T) {
 			},
 			want: "sha256:1acfb4a0eb0acb0b1e8d462df18841581b960d5df190b32f3327df4820e5bc7b",
 		},
+		{
+			name: "empty slice file patterns",
+			args: args{
+				key:     "sha256:5c534be56eca62e756ef2ef51523feda0f19cd7c15bb0c015e3d6e3ae090bf6e",
+				version: "111101112110013",
+			},
+			patterns: []string{},
+			want:     "sha256:1acfb4a0eb0acb0b1e8d462df18841581b960d5df190b32f3327df4820e5bc7b",
+		},
+		{
+			name: "empty string in file patterns",
+			args: args{
+				key:     "sha256:5c534be56eca62e756ef2ef51523feda0f19cd7c15bb0c015e3d6e3ae090bf6e",
+				version: "111101112110013",
+			},
+			patterns: []string{""},
+			want:     "sha256:1acfb4a0eb0acb0b1e8d462df18841581b960d5df190b32f3327df4820e5bc7b",
+		},
+		{
+			name: "multiple empty string in file patterns",
+			args: args{
+				key:     "sha256:5c534be56eca62e756ef2ef51523feda0f19cd7c15bb0c015e3d6e3ae090bf6e",
+				version: "111101112110013",
+			},
+			patterns: []string{"", ""},
+			want:     "sha256:1acfb4a0eb0acb0b1e8d462df18841581b960d5df190b32f3327df4820e5bc7b",
+		},
+		{
+			name: "non empty string in file patterns",
+			args: args{
+				key:     "sha256:5c534be56eca62e756ef2ef51523feda0f19cd7c15bb0c015e3d6e3ae090bf6e",
+				version: "111101112110013",
+			},
+			patterns: []string{"test"},
+			want:     "sha256:7bbca3f56a19cc7d892879d17ba2204f62d37fd9122ca74271e5bb4ab0817bc1",
+		},
+		{
+			name: "empty and non empty string in file patterns",
+			args: args{
+				key:     "sha256:5c534be56eca62e756ef2ef51523feda0f19cd7c15bb0c015e3d6e3ae090bf6e",
+				version: "111101112110013",
+			},
+			patterns: []string{"test", ""},
+			want:     "sha256:7bbca3f56a19cc7d892879d17ba2204f62d37fd9122ca74271e5bb4ab0817bc1",
+		},
+		{
+			name: "empty slice policy paths",
+			args: args{
+				key:     "sha256:5c534be56eca62e756ef2ef51523feda0f19cd7c15bb0c015e3d6e3ae090bf6e",
+				version: "111101112110013",
+			},
+			policy: []string{},
+			want:   "sha256:1acfb4a0eb0acb0b1e8d462df18841581b960d5df190b32f3327df4820e5bc7b",
+		},
+		{
+			name: "empty string in policy paths",
+			skip: "returns unique hash on each run",
+		},
+		{
+			name: "multiple empty string in policy paths",
+			skip: "returns unique hash on each run, different hash from single empty string",
+		},
+		{
```

Comment:
this is not testable as every time i change the expected hash, the resulting hash changes due to that said change.

anyway, all these skips are just for reference, should we omit these?

Reply:
Does "dot" mean the current directory? As you said, it is not testable. We don't need these cases.

Reply:
yes "." as in current dir. okay will remove this then.